lib/gooddata/lcm/actions/provision_clients.rb

```diff
@@ -84,8 +84,15 @@ def call(params)
                       end
                       next
                     end
+
+                    project_id = entry[:project_uri].split('/').last
+                    if collect_synced_status && entry[:status] == 'CREATED' && entry[:id]
+                      # Update project client mappings when there are create new clients during provision clients of the segment
+                      add_project_client_mapping(project_id, entry[:id], segment.segment_id, params)
```

Comment:
For reading purpose, please change to a meaning function name. Eg
```suggestion
                      add_new_clients_to_project_client_mapping(project_id, entry[:id], segment.segment_id, params)
```
Then you can add document to function for details comment/doccumentation

Reply:
Done update code